packages/financial-templates-lib/test/price-feed/InsuredBridgePriceFeed.js

```diff
@@ -0,0 +1,375 @@
+const { InsuredBridgeL2Client } = require("../../dist/clients/InsuredBridgeL2Client");
+
+// Contract objects
+let bridgeAdmin, bridgePool;
+
+// Tested clients
+let pricefeed;
+let l1Client;
+let l2Client;
+let spy;
+
+let finder,
+  store,
+  identifierWhitelist,
+  collateralWhitelist,
+  l1CrossDomainMessengerMock,
+  l2CrossDomainMessengerMock,
+  depositBox,
+  timer,
+  optimisticOracle,
+  l1Token,
+  l2Token,
+  depositData,
+  relayData,
+  depositDataAbiEncoded,
+  depositHash,
+  relayAncillaryData,
+  relayAncillaryDataHash;
+
+// Hard-coded test params:
+const defaultIdentifier = utf8ToHex("IS_CROSS_CHAIN_RELAY_VALID");
+const defaultLiveness = 100;
+const initialPoolLiquidity = toWei("100");
+const relayAmount = toWei("10");
+const defaultProposerBondPct = toWei("0.05");
+const defaultSlowRelayFeePct = toWei("0.01");
+const defaultInstantRelayFeePct = toWei("0.01");
+const lpFeeRatePerSecond = toWei("0.0000015");
+const finalFee = toWei("1");
+const defaultGasLimit = 1_000_000;
+const minimumBridgingDelay = 60; // L2->L1 token bridging must wait at least this time.
+const quoteTimestampOffset = 60; // 60 seconds into the past.
+
+describe("InsuredBridgePriceFeed", function () {
+  let accounts, owner, depositor, relayer, liquidityProvider, recipient;
+
+  const generateRelayParams = (depositDataOverride = {}, relayDataOverride = {}) => {
+    const _depositData = { ...depositData, ...depositDataOverride };
+    const _relayData = { ...relayData, ...relayDataOverride };
+    // Remove the l1Token. This is part of the deposit data (hash) but is not part of the params for relayDeposit.
+    // eslint-disable-next-line no-unused-vars
+    const { l1Token, ...params } = _depositData;
+    return [...Object.values(params), _relayData.realizedLpFeePct];
+  };
+
+  const generateRelayData = async (depositData, relayData, bridgePool) => {
+    // Save other reused values.
+    depositDataAbiEncoded = web3.eth.abi.encodeParameters(
+      ["uint64", "uint64", "address", "address", "address", "uint256", "uint64", "uint64", "uint64"],
+      [
+        depositData.depositId,
+        depositData.depositTimestamp,
+        depositData.recipient,
+        depositData.l2Sender,
+        depositData.l1Token,
+        depositData.amount,
+        depositData.slowRelayFeePct,
+        depositData.instantRelayFeePct,
+        depositData.quoteTimestamp,
+      ]
+    );
+    depositHash = soliditySha3(depositDataAbiEncoded);
+    relayAncillaryData = await bridgePool.methods.getRelayAncillaryData(depositData, relayData).call();
+    relayAncillaryDataHash = soliditySha3(relayAncillaryData);
+    return { depositHash, relayAncillaryData, relayAncillaryDataHash };
+  };
+
+  before(async function () {
+    accounts = await web3.eth.getAccounts();
+    [owner, depositor, relayer, liquidityProvider, recipient] = accounts;
+
+    finder = await Finder.new().send({ from: owner });
```

Comment:
nothing to change right now but I think we should set aside some time to refactor out a lot of this common preamble that a lot of these bridge tests contain.

Reply:
yeah i copied these directly from L1Client and L2Client

Reply:
I'm going to skip this in this PR for the sake of conciseness.